tools/setup_permissions.sh

```diff
@@ -19,81 +20,94 @@ else
-     "${DCOS_URL}/acs/api/v1/acls/dcos:mesos:master:framework:role:${ROLE}/users/${SERVICE_ACCOUNT_NAME}/create"
-
-curl -k -L -X PUT \
-     -H "Authorization: token=${ACS_TOKEN}" \
-     "${DCOS_URL}/acs/api/v1/acls/dcos:mesos:master:framework:role:${ROLE}/users/${SERVICE_ACCOUNT_NAME}/delete"
-
-curl -k -L -X PUT \
-     -H "Authorization: token=${ACS_TOKEN}" \
-     "${DCOS_URL}/acs/api/v1/acls/dcos:mesos:master:task:user:${LINUX_USER}/users/${SERVICE_ACCOUNT_NAME}/create"
-
-curl -k -L -X PUT \
-     -H "Authorization: token=${ACS_TOKEN}" \
-     "${DCOS_URL}/acs/api/v1/acls/dcos:mesos:master:task:user:${LINUX_USER}/users/${SERVICE_ACCOUNT_NAME}/delete"
-
-curl -k -L -X PUT \
-     -H "Authorization: token=${ACS_TOKEN}" \
-     "${DCOS_URL}/acs/api/v1/acls/dcos:mesos:master:reservation:role:${ROLE}/users/${SERVICE_ACCOUNT_NAME}/create"
-
-curl -k -L -X PUT \
-     -H "Authorization: token=${ACS_TOKEN}" \
-     "${DCOS_URL}/acs/api/v1/acls/dcos:mesos:master:reservation:principal:${SERVICE_ACCOUNT_NAME}/users/${SERVICE_ACCOUNT_NAME}/delete"
-
-curl -k -L -X PUT \
-     -H "Authorization: token=${ACS_TOKEN}" \
-     "${DCOS_URL}/acs/api/v1/acls/dcos:mesos:master:volume:role:${ROLE}/users/${SERVICE_ACCOUNT_NAME}/create"
 
-curl -k -L -X PUT \
-     -H "Authorization: token=${ACS_TOKEN}" \
-     "${DCOS_URL}/acs/api/v1/acls/dcos:mesos:master:volume:principal:${SERVICE_ACCOUNT_NAME}/users/${SERVICE_ACCOUNT_NAME}/delete"
 
-echo Permission setup completed for user=$LINUX_USER role=$ROLE
+function grant_registration() {
+    echo "Authorizing ${SERVICE_ACCOUNT_NAME} to register as a Mesos framework with role=${ROLE}"
+
+    curl -k -L -X PUT \
+         -H "Authorization: token=${ACS_TOKEN}" \
+         "${DCOS_URL}/acs/api/v1/acls/dcos:mesos:master:framework:role:${ROLE}" \
+         -d '{"description":"Register with the Mesos master with role=${ROLE}"}' \
+         -H 'Content-Type: application/json'
+    curl -k -L -X PUT \
+         -H "Authorization: token=${ACS_TOKEN}" \
+         "${DCOS_URL}/acs/api/v1/acls/dcos:mesos:master:framework:role:${ROLE}/users/${SERVICE_ACCOUNT_NAME}/create"
+}
+
+function grant_task_execution() {
+    echo "Authorizing ${SERVICE_ACCOUNT_NAME} to execute Mesos tasks as user=${LINUX_USER}"
+    curl -k -L -X PUT \
+         -H "Authorization: token=${ACS_TOKEN}" \
+         "${DCOS_URL}/acs/api/v1/acls/dcos:mesos:master:task:user:${LINUX_USER}" \
+         -d '{"description":"Execute Mesos tasks as user=${LINUX_USER}"}' \
+         -H 'Content-Type: application/json'
+    curl -k -L -X PUT \
+         -H "Authorization: token=${ACS_TOKEN}" \
+         "${DCOS_URL}/acs/api/v1/acls/dcos:mesos:master:task:user:${LINUX_USER}/users/${SERVICE_ACCOUNT_NAME}/create"
+
+    # In order for the Spark Dispatcher to register with Mesos as
+    # root, we must launch the dispatcher task as root.  The other
+    # frameworks are launched as nobody, but then register as
+    # service.user, which defaults to root
```

Comment:
The dispatcher registers as the current linux user.  It's not configurable.  We could make it configurable, but there's really no reason to.